src/main/java/org/jenkinsci/plugins/workflow/steps/WaitForConditionStep.java

```diff
@@ -56,16 +89,19 @@
          */
         private final String id = UUID.randomUUID().toString();
         private static final float RECURRENCE_PERIOD_BACKOFF = 1.2f;
-        static final long MIN_RECURRENCE_PERIOD = 250; // ¼s
-        static final long MAX_RECURRENCE_PERIOD = 15000; // ¼min
-        long recurrencePeriod = MIN_RECURRENCE_PERIOD;
-
-        Execution(StepContext context) {
+        long currentRecurrencePeriod;
```

Comment:
The rename of the `recurrencePeriod` makes the rest of the diff harder to follow.